### ISSUE TITLE
Support retry interval in the retry_policy 

### ISSUE THREAD

**Issue by qiannawang** [envoyproxy/envoy#3940]
Currently, the envoy retry policy [1] provides three following fields:

RetryPolicy {
  "retry_on": "...",
  "num_retries": "...",
  "per_try_timeout_ms" : "..."
}

The per_try_timeout_ms is for hanging TCP connections. If the upstream endpoint was not reachable in TCP, the request from envoy will immediately fail and retries will exhaust immediately, no matter what value we set to num_retries. 

A solution for a real retry to TCP connection failures is to provide a retry_interval in the policy. In this way, envoy will re-send request once every interval configured, if the request fails. 

RetryPolicy{
...
  "retry_interval": "..."
}

Bonus point if we add one more field, enable_multiplier_retry_interval. If this field is set true, the interval waiting for sending the next retry equals to retry_interval * number of retries sent. This would allow the envoy to send retries more aggressively at the beginning and gradually slow down if it retries more times. 

[1] https://www.envoyproxy.io/docs/envoy/latest/api-v1/route_config/route#retry-policy

**Comment by mattklein123** [comment]
> The per_try_timeout_ms is for hanging TCP connections. If the upstream endpoint was not reachable in TCP, the request from envoy will immediately fail and retries will exhaust immediately, no matter what value we set to num_retries.

Sorry I don't follow. Can you rephrase?

I think the issue here is that we don't allow config customization of the retry backoff base value. https://github.com/envoyproxy/envoy/blob/master/source/common/router/retry_state_impl.cc#L74

If this was configurable this would solve your problem, right? I'm pretty sure you still want to use exponential backoff though.
